### PR TITLE
password optional on user edit

### LINKED ISSUE #871: Cannot edit a user without being forced to change their password
When editing a user the password fields should be optional.

### PR DESCRIPTION
**Pull request by AllenBW**
closes #871

### CODE REVIEW

File: src/client/app/components/user-form/user-form.html [hunk 1]
```diff
@@ -67,11 +67,15 @@
...
-                    <input id="user-password" name="password" type="password"
+                    <input id="user-password"
+                           name="password"
+                           type="password"
                            class="user-field__input"
                            ng-model="firstpass"
                            ng-minlength="8"
-                           placeholder="Must be 8 or more characters."  required />
+                           placeholder="Must be 8 or more characters."
+                           ng-required="!vm.editing"
```
Comment (stackus) [inline src/client/app/components/user-form/user-form.html]: How is `vm.editing` being set?

File: src/client/app/components/user-form/user-form.html [hunk 2]
```diff
@@ -84,9 +88,13 @@
...
-                    <input id="user-confirmpassword" ng-model="vm.userToEdit.password"
-                           name="confirmpassword" type="password"
-                           class="user-field__input" placeholder=""  required />
+                    <input id="user-confirmpassword"
+                           ng-model="vm.userToEdit.password"
+                           name="confirmpassword"
+                           type="password"
+                           class="user-field__input"
+                           placeholder=""
+                           ng-required="firstpass"/>
```
Comment (stackus) [inline src/client/app/components/user-form/user-form.html]: Where is `firstpass` coming from?

Comment (AllenBW) [inline src/client/app/components/user-form/user-form.html]: already existed, used for matching password confirmation, didn't touch, but
it doesn't work at the moment

File: src/client/app/components/user-form/user-form.html [hunk 3]
```diff
@@ -67,11 +67,15 @@
                     <label class="user-field__label">Password</label>
                 </div>
                 <div class="user-field__body" ng-class="{'has-error': vm.hasErrors('password')}">
-                    <input id="user-password" name="password" type="password"
+                    <input id="user-password"
+                           name="password"
+                           type="password"
                            class="user-field__input"
                            ng-model="firstpass"
```
Comment (stackus) [inline src/client/app/components/user-form/user-form.html]: Never setting the password onto the model means it'll never be sent to the server.

File: src/client/app/components/user-form/user-form.html (hunk 1 above)
Comment (AllenBW) [inline src/client/app/components/user-form/user-form.html]: directive parameter based on presence or absence of item id, going to
author an issue to clean up this form, that'll be one of the things to fade
away

File: src/client/app/components/user-form/user-form.html (hunk 3 above)
Comment (AllenBW) [inline src/client/app/components/user-form/user-form.html]: quite true, its set in the confirmation

**Pull request merged**